okhttp-tests/src/test/java/okhttp3/internal/framed/Http2Test.java

```diff
@@ -277,10 +277,13 @@ public void pushPromise(int streamId, int promisedStreamId, List<Header> headerB
     frame.writeInt(1);
 
     try {
-      fr.nextFrame(new BaseTestHandler());
-      fail();
+      fr.nextFrame(new BaseTestHandler() {
+        @Override public void settings(boolean clearPrevious, Settings settings) {
+          assert settings.get(7) == 1;
```

Comment:
Oh, good idea. It looks like some of the tests in `Http2Test` don't do that, but I'll do it here. Thanks!